tests/test_serializers.py

```diff
@@ -176,6 +177,12 @@ def check(dataclass, representation, instance):
         check(optional, {'field': {'value': 'A'}}, optional(simple('A')))
         check(optional, {'field': None}, optional(None))
 
+        # New PEP 604 union syntax in Python 3.10+
+        if sys.version_info >= (3, 10, 0):
+            optional = dataclasses.make_dataclass('optional', [('field', simple | None)])
+            check(optional, {'field': {'value': 'A'}}, optional(simple('A')))
+            check(optional, {'field': None}, optional(None))
+
```

Comment:
This isn't necessary, as you already added another check to verify that this syntax gets correctly parsed as being an optional field.